infrastructure/src/main/java/org/corfudb/infrastructure/SequencerServer.java

```diff
@@ -216,20 +206,17 @@ public boolean txnCanCommit(TxResolutionInfo txData) {
                     if (v == null) {
                         return null;
                     }
-                    if (v > txData.getSnapshotTimestamp()) {
-                        log.debug("TX[{}] ABORT due to conflict on stream=[{}] snapshot version=[{}] stream=tail=[{}]",
-                                txData.getTXid(),
-                                Utils.toReadableID(streamID),
-                                txData.getSnapshotTimestamp(),
-                                v);
-                        commit.set(false);
+                    if (v > txInfo.getSnapshotTimestamp()) {
+                        log.debug("ABORT[{}] conflict-stream[{}](ts={})",
+                                txInfo, Utils.toReadableID(streamID), v);
+                        response.set(TokenType.TX_ABORT_CONFLICT.ordinal());
                     }
                     return v;
                 });
             }
         }
 
-        return commit.get();
+        return TokenType.values()[response.get()];
```

Comment:
Do not know how efficient this code construct is. If you use an atomic reference you can return the TokenType directly.